### PR TITLE
replace Open|Fira Sans with Inter; drop woff; use variable fonts

### LINKED ISSUE #10040: Investigate our use of webfonts
A single request to AMO seems to cause 6 different requests to `.woff2` fonts, for a total of 635 KB transferred.
- Can we lower that ? Can we remove some or all of these fonts ?
- Should we even be using those fonts or follow https://protocol.mozilla.org/fundamentals/typography.html instead ?
- Should we [`font-display: swap`](https://developer.mozilla.org/en-US/docs/Web/CSS/@font-face/font-display) to render more quickly without necessarily waiting for the fonts to download ?

### PR DESCRIPTION
**Pull request by eviljeff**
fixes #10040

We currently have 3 types of font - .otf,.woff,.woff2.  WOFF2 format has been supported [since Firefox 39](https://developer.mozilla.org/en-US/docs/Web/Guide/WOFF#browser_compatibility) so offering 2 levels of fallback seemed redundant.  I removed WOFF, leaving just OTF if WOFF2 wasn't supported by the user's browser.  (We could drop OTF too if we thought showing the default sans-serif font was acceptable for unsupported browser versions.)

Though I also added support for [variable fonts](https://developer.mozilla.org/en-US/docs/Web/CSS/CSS_Fonts/Variable_Fonts_Guide) so we could even drop seperate-woff2-file support

### CODE REVIEW

**Comment by willdurand** [comment]
Did we get a sign off from UX?

**Review by bobsilverberg — COMMENTED** [review]
As per the discussion in the issue, I think we need to wait for product and/or UX feedback before moving forward.

**Comment by eviljeff** [comment]
UX have given their blessing to switching to Inter (everywhere)

**Comment by willdurand** [comment]
> Though I also added support for [variable fonts](https://developer.mozilla.org/en-US/docs/Web/CSS/CSS_Fonts/Variable_Fonts_Guide) so we could even drop seperate-woff2-file support

Can you clarify? Do we need WOFF2 or not?

File: README.md
Comment (willdurand) [inline README.md:359]: What's chivo again?

File: src/amo/css/inc/mixins.scss
Comment (willdurand) [inline src/amo/css/inc/mixins.scss:27]: Do we use this mixin? If not, now would be a good time to remove the unused mixins.

File: src/amo/css/inc/mixins.scss
Comment (bobsilverberg) [inline src/amo/css/inc/mixins.scss:18]: Is it possible to make these three lines into a mixin so that they don't have to be duplicated inside the definition of each font mixin below?

File: src/amo/fonts/inter.scss
Comment (bobsilverberg) [inline src/amo/fonts/inter.scss:1]: I notice that we include some `woff2` and `otf` files, e.g. `Inter-ExtraBold` and `Inter-Black` for which there are no definitions in this file. Does that mean those fonts aren't used anywhere on the site, and if so, should we exclude those files? Or does it not really matter than the files are in the repo if they're never referenced in this file?

File: src/amo/fonts/inter.scss
Comment (bobsilverberg) [inline src/amo/fonts/inter.scss:7]: Along the lines of @willdurand's question, do we still need both `woff2` and `otf` fonts?

**Review by bobsilverberg — APPROVED** [review]
I added some questions/comments, but in general it looks good to me.

r+wc

File: README.md
Comment (eviljeff) [reply README.md:359]: heh, no idea, I just removed the two I knew about.  Remove 'chivo' too?

Comment (eviljeff) [reply README.md:359]: Although maybe this entire section is obsolete - do we still bundle old sketch files?

File: src/amo/fonts/inter.scss
Comment (eviljeff) [reply src/amo/fonts/inter.scss:1]: I don't think it matters if they're never referenced.  I included everything in case we were to reference them in the future, but perhaps in the name of tidiness they shouldn't be there?

File: README.md
Comment (bobsilverberg) [reply README.md:359]: I don't think it's been updated in ages, and I don't know if UX even uses it anymore for mockups. I think we can remove this section, but maybe check with @verdi.

File: src/amo/fonts/inter.scss
Comment (eviljeff) [reply src/amo/fonts/inter.scss:7]: (going to aim to answer both questions here).  
So, right now we have otf, .woff, and .woff2 files (all multiple file, non-variable fonts).  A modern browser (Firefox 62+ Chrome 66+) supports all of those, plus a single* file variable font versions of woff2 - though to make things more complicated, variable fonts also have an OS dependency on MacOS (needs 10.13+).  To just support those modern browser+MacOS versions we only need variable woff2 fonts.

There's a decision around what older fonts we should bother supporting as fallback, if any:
 - If none then Firefox will just drops back to a generic sans-serif font on pre 62+macos10.13.
 - If one then we'd get the most coverage by choosing otf, (but it's huge!), or woff2 non-variable (Firefox 39+) for coverage of modern-ish browser/MacOS.
 - Or more than 1.  We have 4 types, so a selection or all 4.

* it's actually 2 files for Inter, because Italic variations are split out

File: README.md
Comment (willdurand) [reply README.md:359]: > do we still bundle old sketch files?

I don't think so, we removed them from the repo a while ago and I don't know where they are 😅

Comment (eviljeff) [reply README.md:359]: It'd be interesting to know from UX what the situation is with invision but I'm not sure it's a relevant link any way - we were done implementing the site mocks a long time ago so now the reference for the frontend should "look" like it the frontend on prod 😄

Comment (willdurand) [reply README.md:359]: I agree, I think we can remove the whole section.

File: src/amo/fonts/inter.scss
Comment (willdurand) [reply src/amo/fonts/inter.scss:7]: > If none then Firefox will just drops back to a generic sans-serif font on pre 62+macos10.13.

According to caniuse, the MacOS version requirement is a Firefox limitation: https://caniuse.com/?search=variable-font

> There's a decision around what older fonts we should bother supporting as fallback, if any:

We should target the browsers listed in the `.browerslist` file, which is: most recent and active browsers (last 2 versions) and _not_ IE. I see that variable fonts aren't supported on UC Browser and Opera Mini but I don't think it's a problem. The MacOS 10.13 requirement is a problem, though, because it is tied to Firefox and it's not a very old MacOS version.

If we keep the patch as it is now, we'd serve up to 2 fonts right? By default, modern browsers would use variable fonts with WOFF2, and the fallback would be OTF?

Comment (eviljeff) [reply src/amo/fonts/inter.scss:7]: the patch as it would offer 3 variants: `woff2` (variable), `woff2` (separate), `otf`.  Maybe we don't need 2 fallbacks though - if we drop `woff` (separate) then older Firefox/MacOS users would get served the much larger OTF font files; if we drop `otf` then super old browser users would see the default sans-serif font.

File: src/amo/fonts/inter.scss
Comment (eviljeff) [reply src/amo/fonts/inter.scss]: https://github.com/mozilla/addons-frontend/pull/10110/commits/485193cf148a192d91ff3492563c6c08acd66674

Comment (eviljeff) [reply src/amo/fonts/inter.scss]: https://github.com/mozilla/addons-frontend/pull/10110/commits/7ff730835fb6130a98bb8fec622dd52637dcabf2

File: src/amo/fonts/inter.scss
Comment (willdurand) [reply src/amo/fonts/inter.scss]: ah ok, so maybe we don't need `woff`? Or we keep things are they are right now and see what we could do in the future? I guess that patch is still an improvement over what we have currently.

File: src/amo/css/inc/mixins.scss
Comment (eviljeff) [reply src/amo/css/inc/mixins.scss:18]: done https://github.com/mozilla/addons-frontend/pull/10110/commits/6ccb26ab7aadde78f42accb1f9e0157ef384e881

File: README.md
Comment (eviljeff) [reply README.md]: dropped the section in https://github.com/mozilla/addons-frontend/pull/10110/commits/f3bbaf86a964b1601a577758523e2c978afdc8d6

**Review by willdurand — APPROVED** [review]

File: src/amo/fonts/inter.scss
Comment (eviljeff) [reply src/amo/fonts/inter.scss]: yeah I think we'll leave this as the patch for now.  Having extra fallbacks won't hurt our performance (only older clients get them) so it's really just about tidiness and less cruft at that point - we can drop non-variable `woff2` or `otf` in a follow-up if we want.

**Pull request merged**
